Marlin/src/HAL/HC32/MarlinHAL.cpp

```diff
@@ -123,6 +123,11 @@ void MarlinHAL::init() {
 
   // Register min serial
   TERN_(POSTMORTEM_DEBUGGING, install_min_serial());
+
+  // warn if low memory after init
+  if (freeMemory() < 1024) {
+    SERIAL_WARN_MSG("HAL: low memory after init!\n");
+  }
```

Comment:
i've added this warning in response to one of @classicrocker883's comments in this PR, where he had issues with random crashes (https://github.com/MarlinFirmware/Marlin/pull/26845#issuecomment-1980218771). 

Besides a memory leak in the sdio driver, i've found that depending on the configuraiton the system could run out of the allocated heap space. 
This warning is meant as a small 'heads-up' message, since out-of-memory issues tend to be really hard to debug.